src/git-command-manager.ts

```diff
@@ -349,34 +360,38 @@ export class GitCommandManager {
     return stdout
   }
 
-  async exec(args: string[], allowAllExitCodes = false): Promise<GitOutput> {
+  async exec(args: string[], { encoding = 'utf8', allowAllExitCodes = false }: ExecOpts = {}): Promise<GitOutput> {
     const result = new GitOutput()
 
     const env = {}
     for (const key of Object.keys(process.env)) {
       env[key] = process.env[key]
     }
 
-    const stdout: string[] = []
-    const stderr: string[] = []
+    const stdout: Buffer[] = []
+    let stdoutLength = 0
+    const stderr: Buffer[] = []
+    let stderrLength = 0
 
     const options = {
       cwd: this.workingDirectory,
       env,
       ignoreReturnCode: allowAllExitCodes,
       listeners: {
         stdout: (data: Buffer) => {
-          stdout.push(data.toString())
+          stdout.push(data)
+          stdoutLength += data.length
         },
         stderr: (data: Buffer) => {
-          stderr.push(data.toString())
+          stderr.push(data)
+          stderrLength += data.length
         }
       }
     }
 
     result.exitCode = await exec.exec(`"${this.gitPath}"`, args, options)
-    result.stdout = stdout.join('')
-    result.stderr = stderr.join('')
+    result.stdout = Buffer.concat(stdout, stdoutLength).toString(encoding)
+    result.stderr = Buffer.concat(stderr, stderrLength).toString(encoding)
```

Comment:
Note these changes are necessary to avoid accidentally incorrectly handling split chunks. In utf-8 this is a problem if the chunk is in the middle of a code point, corrupting the data. In base64 this is a problem because it can end up with padding in the middle of the stream, which effectively truncates the file early.

Reading it all into a single buffer and then converting retains the correct behavior, without letting the executing child process block on a full output stream.